pkg/services/ngalert/state/manager.go

```diff
@@ -30,6 +29,11 @@ type AlertInstanceManager interface {
 	GetStatesForRuleUID(orgID int64, alertRuleUID string) []*State
 }
 
+type StatePersister interface {
+	Async(ctx context.Context, ticker *clock.Ticker, cache *cache)
```

Comment:
Would something like this make it easier to test?

```go
type Result struct {
    Err   err
    Saved []StateTransition
}
```

```suggestion
	Async(ctx context.Context, ticker *clock.Ticker, cache *cache) <-chan Result
```

Reply:
No, I don't think this is needed.